tests/test_phase_psf.py

```diff
@@ -399,57 +436,149 @@ def test_ne():
-            galsim.OpticalScreen(coma2=1.0),
-            galsim.OpticalScreen(trefoil1=1.0),
-            galsim.OpticalScreen(trefoil2=1.0),
-            galsim.OpticalScreen(spher=1.0),
-            galsim.OpticalScreen(spher=1.0, lam_0=100.0),
-            galsim.OpticalScreen(aberrations=[0,0,1.1]), # tip=1.1
+    objs = [galsim.OpticalScreen(diam=1.0, ),
+            galsim.OpticalScreen(diam=1.0, tip=1.0),
+            galsim.OpticalScreen(diam=1.0, tilt=1.0),
+            galsim.OpticalScreen(diam=1.0, defocus=1.0),
+            galsim.OpticalScreen(diam=1.0, astig1=1.0),
+            galsim.OpticalScreen(diam=1.0, astig2=1.0),
+            galsim.OpticalScreen(diam=1.0, coma1=1.0),
+            galsim.OpticalScreen(diam=1.0, coma2=1.0),
+            galsim.OpticalScreen(diam=1.0, trefoil1=1.0),
+            galsim.OpticalScreen(diam=1.0, trefoil2=1.0),
+            galsim.OpticalScreen(diam=1.0, spher=1.0),
+            galsim.OpticalScreen(diam=1.0, spher=1.0, lam_0=100.0),
+            galsim.OpticalScreen(diam=1.0, aberrations=[0,0,1.1]), # tip=1.1
             ]
     all_obj_diff(objs)
 
     # Test PhaseScreenList __ne__
     atm = galsim.Atmosphere(10.0, vx=1.0)
     objs = [galsim.PhaseScreenList(atm),
-            galsim.PhaseScreenList(copy.deepcopy(atm)),  # advance down below
             galsim.PhaseScreenList(objs),  # Reuse list of OpticalScreens above
             galsim.PhaseScreenList(objs[0:2])]
-    objs[1].advance()
     all_obj_diff(objs)
 
     # Test PhaseScreenPSF __ne__
-    objs[0].reset()
     psl = galsim.PhaseScreenList(atm)
-    objs = [galsim.PhaseScreenPSF(psl, 500.0, exptime=0.03, diam=1.0),
-            galsim.PhaseScreenPSF(psl, 500.0, exptime=0.03, diam=1.0)] # advanced so differs
-    psl.reset()
+    objs = [galsim.PhaseScreenPSF(psl, 500.0, exptime=0.03, diam=1.0)]
     objs += [galsim.PhaseScreenPSF(psl, 700.0, exptime=0.03, diam=1.0)]
-    psl.reset()
     objs += [galsim.PhaseScreenPSF(psl, 700.0, exptime=0.03, diam=1.1)]
-    psl.reset()
     objs += [galsim.PhaseScreenPSF(psl, 700.0, exptime=0.03, diam=1.0, flux=1.1)]
-    psl.reset()
     objs += [galsim.PhaseScreenPSF(psl, 700.0, exptime=0.03, diam=1.0, interpolant='linear')]
     stepk = objs[0].stepK()
     maxk = objs[0].maxK()
-    psl.reset()
     objs += [galsim.PhaseScreenPSF(psl, 700.0, exptime=0.03, diam=1.0, _force_stepk=stepk/1.5)]
-    psl.reset()
     objs += [galsim.PhaseScreenPSF(psl, 700.0, exptime=0.03, diam=1.0, _force_maxk=maxk*2.0)]
     all_obj_diff(objs)
 
 
+@timer
+def test_phase_gradient_shoot():
+    # Make the atmosphere
+    seed = 12345
+    r0_500 = 0.2  # m
+    nlayers = 6
+    time_step = 0.025  # s
+    screen_size = 102.4  # m
+    screen_scale = 0.1  # m
+    max_speed = 20  # m/s
+
+    rng = galsim.BaseDeviate(seed)
+    u = galsim.UniformDeviate(rng)
+    Ellerbroek_alts = [0.0, 2.58, 5.16, 7.73, 12.89, 15.46]  # km
+    Ellerbroek_weights = [0.652, 0.172, 0.055, 0.025, 0.074, 0.022]
```

Comment:
Helpful to also add a quick comment saying who Ellerbroek is and why we're using his atmospheric model.